contracts/examples/bonding-curve/src/linear_function.rs

```diff
@@ -0,0 +1,33 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use crate::bc_function::{BCFunction, CurveArguments};
+
+// logic is ax + b  = y
+// x - issued token
+// y - exchanging token
+// a & b - linear equation coeficients
+// because of selling n tokens at a time, the total price y' will be
+// a* (nx + (n - 1) n / 2) + nb = y'
+
+pub struct LinearFunction<BigUint: BigUintApi> {
+	pub a: BigUint,
```

Comment:
Even though the comments explain what "a" and "b" mean, I still think they're bad variable names.
I suggest renaming to something like "linear_coefficient" for "a" and "constant_coefficient" for "b".